src/install/index.js

```diff
@@ -51,8 +51,16 @@ const findDownloadedBuild = deployDoc => {
 };
 
 const downloadBuild = deployDoc => {
-  debug(`Downloading ${keyFromDeployDoc(deployDoc)}, this may take some time…`);
-  return DB.builds.get(keyFromDeployDoc(deployDoc), { attachments: true, binary: true })
+  const docKey = keyFromDeployDoc(deployDoc);
+  debug(`Downloading ${docKey}, this may take some time…`);
+  return DB.builds.get(docKey, { attachments: true, binary: true })
+    .catch(err => {
+      if (err && err.status === 404) {
+        error(`Failed to find a build to download for [${docKey}]. Aborting install.`);
+        return cleanupDeployDoc(deployDoc).reject(err);
```

Comment:
I don't think you want to clean up the old views here - that should only need to happen on successful deploy. If you modify the `cleanupDeployDoc` function to `deleteDeployDoc` then it'll be good to go.